connectors/src/resources/microsoft_resource.ts

```diff
@@ -2,6 +2,9 @@ import type { ModelId, Result } from "@dust-tt/types";
 import { Ok } from "@dust-tt/types";
 import type { Attributes, ModelStatic, Transaction } from "sequelize";
 
+import type { MicrosoftNodeData } from "@connectors/connectors/microsoft/lib/graph_api";
```

Comment:
Not a big fan of creating this strong link between MS API and our resource logic. Can we create our own intermediary type with only the fiels that matter?

Reply:
Agree; intended to refactor post-merge to limit the size of the PR

But fine to do it on this one if you beleive it's better

Reply:
Whatever works the best for you, as long as it's done.